filters/openpolicyagent/tracing.go

```diff
@@ -0,0 +1,47 @@
+package openpolicyagent
+
+import (
+	"net/http"
+
+	opatracing "github.com/open-policy-agent/opa/tracing"
+	"github.com/opentracing/opentracing-go"
+)
+
+func init() {
+	opatracing.RegisterHTTPTracing(&tracingFactory{})
+}
+
+type tracingFactory struct{}
+
+type transport struct {
+	opa     *OpenPolicyAgentInstance
+	wrapped http.RoundTripper
+}
+
+func (*tracingFactory) NewTransport(tr http.RoundTripper, opts opatracing.Options) http.RoundTripper {
+	return &transport{
+		opa:     opts[0].(*OpenPolicyAgentInstance),
+		wrapped: tr,
+	}
+}
+
+func (*tracingFactory) NewHandler(f http.Handler, label string, opts opatracing.Options) http.Handler {
+	return f
+}
+
+func (tr *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	span := opentracing.SpanFromContext(req.Context())
+	ctx := req.Context()
+	if span == nil {
```

Comment:
Seems like span is not used

Reply:
It is used below to inject the headers in the outgoing request.

Reply:
I mean it is overwritten in both branches

Reply:
Ah, got it and cleaned up.